### PR TITLE
Added a loading spinner for requests

### PR DESCRIPTION
**Pull request by apfohl**
I added a loading indicator in the form of a spinner to all request making parts.

### CODE REVIEW

File: components/printer/loading.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package printer
+
+import (
+	"time"
+
+	"github.com/briandowns/spinner"
+)
+
+type Func[T any] func() (T, error)
```
Comment (Kharonus) [inline components/printer/loading.go]: do you intent to use this type elsewhere? If not, I'd rather not export it.

File: components/printer/loading.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+var loadingSpinner *spinner.Spinner
+
+func init() {
```
Comment (Kharonus) [inline components/printer/loading.go:13]: Who calls this `init` function?

File: components/requests/requests.go [hunk 3]
```diff
@@ -25,15 +26,21 @@ func Init(hostUrl *url.URL, tokenValue string) {
...
 }
 
 func Patch(path string, requestBody *RequestData) (responseBody []byte, err error) {
-	return Do("PATCH", path, nil, requestBody)
+	loadingFunc := func() ([]byte, error) { return Do("PATCH", path, nil, requestBody) }
+
+	return printer.WithSpinner(loadingFunc)
 }
 
 func Do(method string, path string, query *Query, requestData *RequestData) (responseBody []byte, err error) {
```
Comment (Kharonus) [inline components/requests/requests.go:46]: I think, we should reduce visibility of the `Do` function to `do`, then we can assure, that every requests uses the spinner.

the 100ms used in `loading.go`, is that the debounce?

**Review by Kharonus — CHANGES_REQUESTED** [review]
Nice work, I'd just like to get in a few changes.

File: components/requests/requests.go (hunk 3 above)
Comment (apfohl) [reply components/requests/requests.go:46]: By go itself: https://stackoverflow.com/questions/24790175/when-is-the-init-function-run

File: components/printer/loading.go (hunk 2 above)
Comment (apfohl) [reply components/printer/loading.go:13]: By go itself: https://stackoverflow.com/questions/24790175/when-is-the-init-function-run

File: components/requests/requests.go (hunk 3 above)
Comment (apfohl) [reply components/requests/requests.go:46]: > I think, we should reduce visibility of the Do function to do, then we can assure, that every requests uses the spinner.

I agree.

The 100ms is the animation delay between the pictures.

File: components/printer/loading.go (hunk 1 above)
Comment (apfohl) [reply components/printer/loading.go]: It can be private indeed. I just couldn't leave it out.

**Review by Kharonus — APPROVED** [review]

**Pull request merged**